packages/core/contracts/financial-templates/common/financial-product-libraries/KpiOptionsFinancialProductLibrary.sol

```diff
@@ -0,0 +1,49 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+import "./FinancialProductLibrary.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../../../common/implementation/Lockable.sol";
+
+/**
+ * @title KPI Options Financial Product Library
+ * @notice Adds custom tranformation logic to modify the price and collateral requirement behavior of the expiring multi party contract.
+ * If a price request is made pre-expiry, the price should always be set to 2 and the collateral requirement should be set to 1.
+ * Post-expiry, the collateral requirement is left as 1 and the price is left unchanged.
+ */
+contract KpiOptionsFinancialProductLibrary is FinancialProductLibrary, Ownable, Lockable {
```

Comment:
Is it possible in the financial product library framework to deploy a generalized version of this, that would map EMP addresses to pre-expiry prices (in this case, `2`), to save on gas if a single entity was deploying multiple KPI options? If not, is that because deployed EMP addresses are non-deterministic, and maybe this would be possible with CREATE2?

Reply:
> If not, is that because deployed EMP addresses are non-deterministic, and maybe this would be possible with CREATE2?

There is probably a way around this, by creating a mapping of `addressees => prices` and modifying the factory code to take in an array of prices to set for deployed contracts and deploying new contracts with these prices.